libraries/matrix/impl/src/main/kotlin/io/element/android/libraries/matrix/impl/room/JoinedRustRoom.kt

```diff
@@ -136,11 +136,21 @@ class JoinedRustRoom(
 
     override val roomNotificationSettingsStateFlow = MutableStateFlow<RoomNotificationSettingsState>(RoomNotificationSettingsState.Unknown)
 
-    override val liveTimeline = liveInnerTimeline.map(mode = Timeline.Mode.Live) {
-        syncUpdateFlow.value = systemClock.epochMillis()
-    }
+    override val liveTimeline = liveInnerTimeline.map(mode = Timeline.Mode.Live)
+
+    override val syncUpdateFlow = liveTimeline.onSyncedEventReceived
+        .map { systemClock.epochMillis() }
```

Comment:
This is not new but maybe source of issue. Maybe replace the time by a simple counter?
```kotlin
    private var counter: Long = 0L
    override val syncUpdateFlow = liveTimeline.onSyncedEventReceived
        .map { ++counter }
        .stateIn(
            scope = roomCoroutineScope,
            started = WhileSubscribed(),
            initialValue = 0L,
        )
```